python/dgl/sparse/diag_matrix.py

```diff
@@ -8,23 +8,7 @@
 
 
 class DiagMatrix:
-    """Diagonal Matrix Class
-
-    Parameters
-    ----------
-    val : torch.Tensor
-        Diagonal of the matrix. It can take shape (N) or (N, D).
-    shape : tuple[int, int], optional
-        If not specified, it will be inferred from :attr:`val`, i.e.,
-        (N, N). Otherwise, :attr:`len(val)` must be equal to :attr:`min(shape)`.
-
-    Attributes
-    ----------
-    val : torch.Tensor
-        Diagonal of the matrix.
-    shape : tuple[int, int]
-        Shape of the matrix.
-    """
```

Comment:
These should be recovered. See https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes .

Reply:
I am removing this since this is not the concrete list of attributes, and we also did not doc SparseMatrix.
If needed, we should add all Attributes for both SparseMatrix and DiagMatrix.
My feeling is the list could be too long to list here which makes the doc string unreadable, and we don't have a rst file relying on this.

Reply:
If you believe it is necessary to list all the Attributes, let's do it properly in a separated PR.

Reply:
Oh, I'm not talking about attributes, but the docstring of `__init__` (i.e., the parameter section).

Reply:
Added.